crates/payload/builder/src/service.rs

```diff
@@ -271,10 +277,11 @@ where
     }
 }
 
-impl<Gen> Future for PayloadBuilderService<Gen>
+impl<Gen, St> Future for PayloadBuilderService<Gen, St>
 where
     Gen: PayloadJobGenerator + Unpin + 'static,
     <Gen as PayloadJobGenerator>::Job: Unpin + 'static,
+    St: Stream<Item = CanonStateNotification> + Send + Unpin + 'static,
 {
```

Comment:
here' we need to poll the chain_events and then call the generator with the event.

The `on_new_state` we need to extract the new `tip` and extract the state+account into an initial CachedReads